### PR TITLE
BinaryPropagationHandler split between interface and implementation.

### PR DESCRIPTION
**Pull request by bogdandrutu**
This is a preparation for https://github.com/google/instrumentation-java/pull/201

### CODE REVIEW

File: core/src/main/java/com/google/instrumentation/trace/BinaryPropagationHandler.java [hunk 1]
```diff
@@ -18,55 +18,55 @@
...
+ * <pre>{@code
+ * private static final Tracer tracer = TraceService.getTracer();
+ * private static final BinaryPropagationHandler binaryPropagationHandler =
+ *     TraceService.getBinaryPropagationHandler();
+ * void onRequestReceived() {
+ *   // Get the binaryValue from the request.
+ *   SpanContext spanContext = SpanContext.INVALID;
+ *   try {
+ *     if (binaryValue != null) {
+ *       binaryPropagationHandler.fromBinaryValue(binaryValue);
```
Comment (sebright) [inline core/src/main/java/com/google/instrumentation/trace/BinaryPropagationHandler.java]: This line should set spanContext.

File: core/src/test/java/com/google/instrumentation/trace/BinaryPropagationHandlerTest.java [hunk 2]
```diff
@@ -14,179 +14,33 @@
...
-import org.junit.Rule;
 import org.junit.Test;
-import org.junit.rules.ExpectedException;
-import org.junit.runner.RunWith;
-import org.junit.runners.JUnit4;
-import org.mockito.Mock;
-import org.mockito.MockitoAnnotations;
 
 /** Unit tests for {@link BinaryPropagationHandler}. */
-@RunWith(JUnit4.class)
```
Comment (sebright) [inline core/src/test/java/com/google/instrumentation/trace/BinaryPropagationHandlerTest.java]: Should this line be added back?

File: core/src/main/java/com/google/instrumentation/trace/BinaryPropagationHandler.java (hunk 1 above)
Comment (bogdandrutu) [inline core/src/main/java/com/google/instrumentation/trace/BinaryPropagationHandler.java]: Done.

File: core/src/test/java/com/google/instrumentation/trace/BinaryPropagationHandlerTest.java (hunk 2 above)
Comment (bogdandrutu) [inline core/src/test/java/com/google/instrumentation/trace/BinaryPropagationHandlerTest.java]: Done.

**Pull request merged**